rasa/shared/utils/schemas/markers.py

```diff
@@ -0,0 +1,56 @@
+MARKER_ID_SCHEMA = {
+    "description": "The unique identifier for a marker.",
+    "type": "string",
+}
+
+OPERATOR_SCHEMA = {
+    "description": "The operator to apply to the list of conditions.",
+    "type": "string",
+    "default": "AND",
+    "enum": ["AND", "OR", "SEQ"],
+}
+
+EVENT_SCHEMA = {
+    "description": "A list of events nested under a condition.",
+    "type": "array",
+    "items": {"type": "string"},
+    "minItems": 1,
+    "uniqueItems": True,
+}
+
+CONDITION_SCHEMA = {
+    "description": "The list of conditions for a marker",
+    "type": "array",
+    "items": {
+        "type": "object",
+        "properties": {
+            "slot_is_set": EVENT_SCHEMA,
+            "slot_is_not_set": EVENT_SCHEMA,
+            "action_executed": EVENT_SCHEMA,
+            "action_not_executed": EVENT_SCHEMA,
+            "intent_detected": EVENT_SCHEMA,
+            "intent_not_detected": EVENT_SCHEMA,
+            "user_uttered": EVENT_SCHEMA,
+            "bot_uttered": EVENT_SCHEMA,
+        },
+    },
```

Comment:
Should this also specify `minItems: 1` if you expect at least 1 dict in the array?